### PR TITLE
SOUS-1063 use sous server from SOUS_SERVER

### PR DESCRIPTION
**Pull request by JC1738**

### CODE REVIEW

**Comment by sous-machine-account** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: graph/actions.go [hunk 1]
```diff
@@ -193,15 +193,26 @@ func (di *SousGraph) GetDeploy(opts DeployActionOpts) (actions.Action, error) {
...
 		Config           LocalSousConfig
+		TraceID          sous.TraceID
 	}{}
 	if err := di.Inject(&scoop); err != nil {
 		return nil, err
 	}
 	rf := (*sous.ResolveFilter)(scoop.ResolveFilter)
+
+	client := scoop.HTTP.HTTPClient
+	if len(os.Getenv("SOUS_USE_SOUS_SERVER")) > 0 {
```
Comment (nyarly) [inline graph/actions.go]: Prefer using LookupEnv to test if an environment variable is set or not.

File: graph/actions.go [hunk 2]
```diff
@@ -193,15 +193,26 @@ func (di *SousGraph) GetDeploy(opts DeployActionOpts) (actions.Action, error) {
...
+		c, err := restful.NewClient(scoop.Config.Config.Server, scoop.LogSink.LogSink.Child(opts.DFF.Cluster+".http-client"), map[string]string{"OT-RequestId": string(scoop.TraceID)})
+		if err != nil {
+			return nil, err
+		}
+		client = c
```
Comment (nyarly) [inline graph/actions.go]: I'd do `var err error` and use normal assignment. Probably just a matter of idiom though.

File: graph/actions.go (hunk 1 above)
Comment (JC1738) [reply graph/actions.go]: Cool, didn't know that existed, will change.

File: graph/actions.go (hunk 2 above)
Comment (JC1738) [reply graph/actions.go]: Yes, I went back and forth on that

**Comment by JC1738** [comment]
@samsalisbury This is passing on Jenkins, as well as deployed to ot-sous-server-dev (tested builds and deploys there with it)

**Pull request merged**